File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/producers/OptionsBasedProducer.java
package ru.kmorozov.gbd.core.producers;

import org.apache.commons.lang3.StringUtils;
import ru.kmorozov.gbd.core.config.GBDOptions;
import ru.kmorozov.gbd.core.logic.context.ContextProvider;
import ru.kmorozov.gbd.core.logic.context.IBookListProducer;
import ru.kmorozov.gbd.core.logic.library.LibraryFactory;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

/**
 * Created by km on 12.11.2016.
 */
public class OptionsBasedProducer implements IBookListProducer {

    private Set<String> bookIdsList = new HashSet<>();

    public OptionsBasedProducer() {
        String bookId = GBDOptions.getBookId();

        if (!StringUtils.isEmpty(bookId) && LibraryFactory.isValidId(bookId))
            this.bookIdsList = new HashSet<>(Collections.singletonList(bookId));
        else if (GBDOptions.isValidConfig())
            this.bookIdsList = ContextProvider.getContextProvider().getBookIdsList();
    }

    @Override
    public Set<String> getBookIds() {
        return this.bookIdsList;
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/producers/OptionsBasedProducer.kt
package ru.kmorozov.gbd.core.producers

import com.google.common.base.Strings
import ru.kmorozov.gbd.core.config.GBDOptions
import ru.kmorozov.gbd.core.logic.context.ContextProvider
import ru.kmorozov.gbd.core.logic.context.IBookListProducer
import ru.kmorozov.gbd.core.logic.library.LibraryFactory
import java.util.*

/**
 * Created by km on 12.11.2016.
 */
class OptionsBasedProducer : IBookListProducer {

    override var bookIds: Set<String> = HashSet()
        private set

    init {
        val bookId = GBDOptions.bookId

        if (!Strings.isNullOrEmpty(bookId))
            bookIds = bookId.split(";").filter { LibraryFactory.isValidId(it) }.toSet()
        else if (GBDOptions.isValidConfig)
            bookIds = ContextProvider.contextProvider.bookIdsList
    }
}
